internal/common/autoscaler/types.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2017-2021 Uber Technologies Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package autoscaler
+
+type (
+	// Resource is the unit of scalable resources
+	Resource uint
+
+	// UsageInMilli is the unit of Resource usage times 1000
+	UsageInMilli uint64
```

Comment:
And just to clarify the naming of this - its got nothing to do with milliseconds, right? As far as I can tell, its just resource usage * 1000. Milli in the name makes it sound like milliseconds, but maybe thats just me. Would this not work as a float?

Reply:
I've changed to MilliUsage similar to how kubernetes autoscaler named Millicores. https://sourcegraph.com/github.com/kubernetes/autoscaler@master/-/blob/vertical-pod-autoscaler/pkg/recommender/logic/recommender.go#L27:5 

I was trying to avoid using float to avoid negative numbers from the beginning. Otherwise, I'll need to handle negative cases in a lot of places unnecessarily.